index.js

```diff
@@ -91,16 +123,20 @@ module.exports = function renderDom(rootPostOrString, options, cb) {
 			resetPartial: resetPartial
 		}
 
-		butler.on('post changed', function (filename, post) {
+		butler.on('post changed', function(filename, post) {
 			if (partialExists(filename)) { // Only cares about posts that are in the system
 				if (filename === currentFilename) { // Current post changed
 					setCurrent(post)
 				} else {
 					scan(post, util, state, true)
 				}
 			}
+
+			posts[filename] = post
```

Comment:
Is there a reason this line isn't between lines 126 and 127? (Before the `if (partialExists(filename))` line)

Then we might not need line 92.